### PR TITLE
Update ICTR data to more closely reflect real data

### LINKED ISSUE #169: In ICTR use new data

### PR DESCRIPTION
**Pull request by kalashsinghal**
closes #169 

Dear @maurolepore, 

I have put the changed data in a different folder as of now which I will change in later commits. This PR is to check the initial test failures due to changes in the data structure. Please feel free to add your suggestions! 

FYI: As `activity_uuid_product_uuid` is the unique identifier, I have removed `ei_activity` and `unit` for joining the dataframes after consulting with Tilman. Also changed the tests accordingly.

### CODE REVIEW

File: data/ictr_inputs.rda
Comment (maurolepore) [inline data/ictr_inputs.rda:1]: Same as above.

File: data-raw/ictr_data.R [hunk 1]
```diff
@@ -1,12 +1,12 @@
...
 library(usethis)
 
 # The raw data comes from tiltIndicator#87, which includes pre-processing
-ictr_inputs <- read_csv(here("data-raw", "ictr", "ictr_inputs.csv"), show_col_types = FALSE) |>
+ictr_inputs <- vroom(here("data-raw", "ictr", "real_data", "ictr_inputs.csv"), show_col_types = FALSE) |>
   select(all_of(ictr_inputs_crucial()))
 use_data(ictr_inputs, overwrite = TRUE)
 
-ictr_companies <- read_csv(here("data-raw", "ictr", "ictr_companies.csv"), show_col_types = FALSE) |>
+ictr_companies <- vroom(here("data-raw", "ictr", "real_data", "ictr_companies.csv"), show_col_types = FALSE) |>
```
Comment (maurolepore) [inline data-raw/ictr_data.R:12]: Why do we change to vroom?

File: data-raw/ictr_data.R [hunk 2]
```diff
@@ -1,12 +1,14 @@
+library(vroom)
 library(here)
-library(readr)
 library(usethis)
+library(dplyr)
+devtools::load_all()
```
Comment (maurolepore) [inline data-raw/ictr_data.R:5]: FYI I think we need dplyr to use `select()` and `devtools::load_all()` to use `ictr_inputs_crucial()`, right?

File: data-raw/ictr/real_data/ictr_inputs.csv
Comment (maurolepore) [inline data-raw/ictr/real_data/ictr_inputs.csv:1]: Why do we preserve the old files?

We are overwriting the data/ so we may as well overwrite the data-raw/. 

If we can, it would be best to delete the old raw dataset now. Later it'll be harder to know what it's there and we'll be afraid to delete it so it'll likely hang in there forever.

File: DESCRIPTION [hunk 3]
```diff
@@ -33,7 +33,8 @@ Suggests:
     purrr,
     rmarkdown,
     roxygen2,
-    testthat (>= 3.0.0)
+    testthat (>= 3.0.0),
+    usethis
```
Comment (maurolepore) [inline DESCRIPTION:37]: FYI I added usethis to `Suggests:`. We should probabbly do the same with readr and vroom (but first let's discuss if we need vroom or if readr is enough).

**Review by maurolepore — APPROVED** [review]
LGTM with comments.

You may address my comments and merge. 

Thanks!

File: data-raw/ictr_data.R (hunk 1 above)
Comment (kalashsinghal) [reply data-raw/ictr_data.R:12]: @maurolepore I experienced delimiting issues while reading the new csv file using `read_csv`, hence I used vroom. I tried many other options with `read_csv` for reading the new csv file but none worked. You can check from your end as well if `read_csv` works for you if you change the column names of existing csv file. If it will work for you then we can have a meeting to discuss my case and decide what to use.

File: data-raw/ictr/real_data/ictr_inputs.csv
Comment (kalashsinghal) [reply data-raw/ictr/real_data/ictr_inputs.csv:1]: @maurolepore I have deleted the old files as suggested by you :)

File: data-raw/ictr_data.R (hunk 1 above)
Comment (maurolepore) [reply data-raw/ictr_data.R]: I can check later today. I suspect the issue relates to [this new guideline](https://gist.github.com/maurolepore/17b468dd8c3a603c668220563754ad7f#in-csv-files-use-the-global-format).

Comment (kalashsinghal) [reply data-raw/ictr_data.R]: @maurolepore I have read the guideline and tried different .csv formats to save the changes but it didn't work out. Please let me know if you want to show me the solution in a meeting! Thanks!

Comment (kalashsinghal) [reply data-raw/ictr_data.R]: @maurolepore  I tried to follow the guideline but it didn't work out due to an error. Please check from your side and let me know what worked for you! Thanks!

Comment (maurolepore) [reply data-raw/ictr_data.R]: Did  you try `read_csv2()`?

I see that `vroom()` guesses the correct delimiter but fails to understand the correct decimal point.

``` r
path <- here::here("data-raw", "ictr", "ictr_inputs.csv")

# Good
readr::read_csv2(path, col_select = "input_co2", , col_types = "d")
#> ℹ Using "','" as decimal and "'.'" as grouping mark. Use `read_delim()` for more control.
#> # A tibble: 75 × 1
#>    input_co2
#>        <dbl>
#>  1      73.6
#>  2      96  
#>  3      39.1
#>  4      80.1
#>  5      10.7
#>  6      25.7
#>  7      16.9
#>  8      55.8
#>  9      29.1
#> 10      92.7
#> # ℹ 65 more rows

# Bad: Uses the correct delimiter but the wrong decimal point
vroom::vroom(path, col_select = "input_co2", col_types = "d")
#> # A tibble: 75 × 1
#>    input_co2
#>        <dbl>
#>  1      7358
#>  2        96
#>  3      3907
#>  4      8012
#>  5      1072
#>  6      2571
#>  7      1693
#>  8      5582
#>  9      2913
#> 10      9274
#> # ℹ 65 more rows
```

Can you read the data with `read_csv2()` then rewrite it as the new raw data? Then read it with `read_csv()`.

Try one more time else I'll take over.

File: data-raw/ictr_data.R [hunk 4]
```diff
@@ -3,10 +3,8 @@ library(readr)
 library(usethis)
 
 # The raw data comes from tiltIndicator#87, which includes pre-processing
```
Comment (maurolepore) [inline data-raw/ictr_data.R:5]: @kalashsinghal, is this comment still valid? 
Where did you get this latest version of the example datasets?

File: data-raw/ictr_data.R [hunk 5]
```diff
@@ -3,10 +3,8 @@ library(readr)
...
-ictr_inputs <- read_csv(here("data-raw", "ictr", "ictr_inputs.csv"), show_col_types = FALSE) |>
-  select(all_of(ictr_inputs_crucial()))
+ictr_inputs <- read_csv(here("data-raw", "ictr", "ictr_inputs.csv"))
 use_data(ictr_inputs, overwrite = TRUE)
 
-ictr_companies <- read_csv(here("data-raw", "ictr", "ictr_companies.csv"), show_col_types = FALSE) |>
```
Comment (maurolepore) [inline data-raw/ictr_data.R:10]: The guidelines worked for me. I read then re-wrote the few columns we care about. This way the raw data is lighter and also the code to read it is much simpler.

I intentionally want to see the messages from `read_csv()` to make sure it's using the correct delimiter, decimal point, and column types.

**Review by maurolepore — APPROVED** [review]
LGTM

File: R/ictr_score_companies.R [hunk 6]
```diff
@@ -41,7 +41,7 @@ ictr_score_companies <- function(ecoinvent_scores, companies) {
   ## join by activity_product_uuid and other joint columns from companies with
   ## ecoinvent_scores
   companies_scores <- companies |>
-    left_join(ecoinvent_scores, by = c("activity_product_uuid", "ei_activity", "unit"))
+    left_join(ecoinvent_scores, by = c("activity_uuid_product_uuid"))
```
Comment (maurolepore) [inline R/ictr_score_companies.R:44]: Mmm ... this doesn't belong to this PR.

Comment (maurolepore) [reply R/ictr_score_companies.R:44]: See https://github.com/2DegreesInvesting/tiltIndicator/pull/179

File: data-raw/ictr_data.R (hunk 5 above)
Comment (kalashsinghal) [reply data-raw/ictr_data.R:10]: @maurolepore Renaming the columns using `rename` function worked at the end and I could finally be able to read the updated csv file with `read_csv` function. Btw the `read_csv2` also did not work if I update the csv file manually. 

Interesting fact: I can now read the csv files (you uploaded in your recent commits) with `read_csv` function even if I make the changes in the file manually. This is some weird behaviour of csv files I don't understand! Nevertheless, it's good at the end that the problem is solved. :)

File: data-raw/ictr_data.R (hunk 4 above)
Comment (kalashsinghal) [reply data-raw/ictr_data.R]: @maurolepore changed it to this PR tiltIndicator #170 (given by Tilman on slack)
